bug: No file openssl/ssl.h in /usr/local/openresty/openssl/include

### Issue description

### Environment

Bug report without environment information will be ignored or closed.

* apisix version (cmd: `apisix version`):
* OS (cmd: `uname -a`): 4.15.0-137-generic #141-Ubuntu SMP Fri Feb 19 13:46:27 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
* OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`): openresty/1.19.3.1
* etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API): 404 page not found
* apisix-dashboard version, if have:
* luarocks version, if the issue is about installation (cmd: `luarocks --version`):

### Minimal test code / Steps to reproduce the issue

Bug report without steps to reproduce will be ignored or closed.

1.
2.
3.

### What's the actual result? (including assertion message & call stack if applicable)

### What's the expected result?

make deps
```
/build/apisix-2.7$ make deps 
mkdir -p ~/.luarocks
luarocks config --local variables.OPENSSL_LIBDIR /usr/local/openresty/openssl/lib
Wrote
	variables.OPENSSL_LIBDIR = "/usr/local/openresty/openssl/lib"
to
	/home/www/.luarocks/config-5.1.lua
luarocks config --local variables.OPENSSL_INCDIR /usr/local/openresty/openssl/include
Wrote
	variables.OPENSSL_INCDIR = "/usr/local/openresty/openssl/include"
to
	/home/www/.luarocks/config-5.1.lua
luarocks install rockspec/apisix-master-0.rockspec --tree=deps --only-deps --local 
Missing dependencies for apisix master-0:
   luasec 0.9-1 (not installed)
   lua-resty-consul 0.3-2 (not installed)
   penlight 1.9.2-1 (not installed)
   ext-plugin-proto 0.1.1 (not installed)

apisix master-0 depends on lua-resty-ctxdump 0.1-0 (0.1-0 installed)
apisix master-0 depends on lua-resty-dns-client 5.2.0 (5.2.0-1 installed)
apisix master-0 depends on lua-resty-template 2.0 (2.0-1 installed)
apisix master-0 depends on lua-resty-etcd 1.5.0 (1.5.0-0 installed)
apisix master-0 depends on lua-resty-balancer 0.02rc5 (0.02rc5-0 installed)
apisix master-0 depends on lua-resty-ngxvar 0.5.2 (0.5.2-0 installed)
apisix master-0 depends on lua-resty-jit-uuid 0.0.7 (0.0.7-2 installed)
apisix master-0 depends on lua-resty-healthcheck-api7 2.2.0 (2.2.0-0 installed)
apisix master-0 depends on lua-resty-jwt 0.2.0 (0.2.0-0 installed)
apisix master-0 depends on lua-resty-hmac-ffi 0.05 (0.05-0 installed)
apisix master-0 depends on lua-resty-cookie 0.1.0 (0.1.0-1 installed)
apisix master-0 depends on lua-resty-session 2.24 (2.24-1 installed)
apisix master-0 depends on opentracing-openresty 0.1 (0.1-0 installed)
apisix master-0 depends on lua-resty-radixtree 2.8.0 (2.8.0-0 installed)
apisix master-0 depends on lua-protobuf 0.3.2 (0.3.2-0 installed)
apisix master-0 depends on lua-resty-openidc 1.7.2-1 (1.7.2-1 installed)
apisix master-0 depends on luafilesystem 1.7.0-2 (1.7.0-2 installed)
apisix master-0 depends on lua-tinyyaml 1.0 (1.0-0 installed)
apisix master-0 depends on nginx-lua-prometheus 0.20210206 (0.20210206-1 installed)
apisix master-0 depends on jsonschema 0.9.5 (0.9.5-0 installed)
apisix master-0 depends on lua-resty-ipmatcher 0.6 (0.6-0 installed)
apisix master-0 depends on lua-resty-kafka 0.07 (0.07-0 installed)
apisix master-0 depends on lua-resty-logger-socket 2.0-0 (2.0-0 installed)
apisix master-0 depends on skywalking-nginx-lua 0.4-1 (0.4-1 installed)
apisix master-0 depends on base64 1.5-2 (1.5-2 installed)
apisix master-0 depends on binaryheap 0.4 (0.4-1 installed)
apisix master-0 depends on dkjson 2.5-2 (2.5-2 installed)
apisix master-0 depends on resty-redis-cluster 1.02-4 (1.02-4 installed)
apisix master-0 depends on lua-resty-expr 1.3.0 (1.3.0-0 installed)
apisix master-0 depends on graphql 0.0.2 (0.0.2-1 installed)
apisix master-0 depends on argparse 0.7.1-1 (0.7.1-1 installed)
apisix master-0 depends on luasocket 3.0rc1-2 (3.0rc1-2 installed)
apisix master-0 depends on luasec 0.9-1 (not installed)
Installing https://luarocks.org/luasec-0.9-1.src.rock

Error: Failed installing dependency: https://luarocks.org/luasec-0.9-1.src.rock - Could not find header file for OPENSSL
  No file openssl/ssl.h in /usr/local/openresty/openssl/include
You may have to install OPENSSL in your system and/or pass OPENSSL_DIR or OPENSSL_INCDIR to the luarocks command.
Example: luarocks install luasec OPENSSL_DIR=/usr/local
Makefile:80: recipe for target 'deps' failed
make: *** [deps] Error 1

```

Unlikely, I got this trouble :-(

1. I had installed (via RPM repository)[https://github.com/apache/apisix/blob/master/docs/en/latest/how-to-build.md#installation-via-rpm-repositorycentos-7], this got `openresty` and `apache apisix`, like:

```bash
[vagrant@lvs apisix]$ rpm -aq|grep apisix
apisix-base-1.19.9.1.3-0.el7.x86_64
apache-apisix-repo-1.0-1.noarch
apisix-2.12.0-0.el7.x86_64
[vagrant@lvs apisix]$ rpm -aq|grep openresty
openresty-pcre-8.44-1.el7.x86_64
openresty-zlib-1.2.11-3.el7.centos.x86_64
openresty-openssl111-1.1.1l-1.el7.x86_64
```

2. I cloned apisix source and `make deps`, logs lay [here](https://gist.github.com/kwanhur/68fc612700294e3b48a43d57a3beb1d1) got failure:

```bash
apisix master-0 depends on luasec 0.9-1 (not installed)
Installing https://luarocks.org/luasec-0.9-1.src.rock

Error: Failed installing dependency: https://luarocks.org/luasec-0.9-1.src.rock - Could not find header file for OPENSSL
  No file openssl/ssl.h in /usr/local/openresty/openssl111/include
You may have to install OPENSSL in your system and/or pass OPENSSL_DIR or OPENSSL_INCDIR to the luarocks command.
Example: luarocks install luasec OPENSSL_DIR=/usr/local
make: *** [deps] Error 1
```
3. I did `./utils/install-dependencies.sh`, unfortunately, install `$common_dep` failed:

- existed `apache-apisix-repo-1.0-1.noarch` conflicts with this installation

```bash
[vagrant@lvs apisix]$ ./utils/install-dependencies.sh
+ main
++ uname -s
++ tr '[:upper:]' '[:lower:]'
+ OS_NAME=linux
+ [[ 0 == 0 ]]
+ [[ linux == \l\i\n\u\x ]]
+ multi_distro_installation
+ grep -Eqi CentOS /etc/issue
+ grep -Eq CentOS /etc/centos-release /etc/os-release /etc/redhat-release /etc/system-release
+ install_dependencies_with_yum centos
+ sudo yum install -y yum-utils
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Package yum-utils-1.1.31-54.el7_8.noarch already installed and latest version
Nothing to do
+ local 'common_dep=curl git gcc openresty-openssl111-devel unzip pcre pcre-devel openldap-devel'
+ '[' centos == centos ']'
+ sudo yum install -y https://repos.apiseven.com/packages/centos/apache-apisix-repo-1.0-1.noarch.rpm
Loaded plugins: fastestmirror
apache-apisix-repo-1.0-1.noarch.rpm                                                                                                    | 2.6 kB  00:00:00
Examining /var/tmp/yum-root-arpfF4/apache-apisix-repo-1.0-1.noarch.rpm: apache-apisix-repo-1.0-1.noarch
/var/tmp/yum-root-arpfF4/apache-apisix-repo-1.0-1.noarch.rpm: does not update installed package.
Error: Nothing to do
```

solution: modify `utils/install-dependencies.sh` ignore installed package `apache-apisix-repo-1.0-1.noarch` and redo `./utils/install-dependencies.sh multi_distro_installation`, then `make deps` get off this trouble

```bash
[vagrant@lvs apisix]$ git diff utils/install-dependencies.sh
diff --git a/utils/install-dependencies.sh b/utils/install-dependencies.sh
index 3662326..b152bb2 100755
--- a/utils/install-dependencies.sh
+++ b/utils/install-dependencies.sh
@@ -49,7 +49,7 @@ function install_dependencies_with_yum() {
     local common_dep="curl git gcc openresty-openssl111-devel unzip pcre pcre-devel openldap-devel"
     if [ "${1}" == "centos" ]; then
         # add APISIX source
-        sudo yum install -y https://repos.apiseven.com/packages/centos/apache-apisix-repo-1.0-1.noarch.rpm
+        rpm -q --quiet apache-apisix-repo-1.0-1.noarch || sudo yum install -y https://repos.apiseven.com/packages/centos/apache-apisix-repo-1.0-1.noarch.rpm

         # install apisix-base and some compilation tools
         # shellcheck disable=SC2086
```

--- a/utils/install-dependencies.sh
+++ b/utils/install-dependencies.sh
@@ -49,7 +49,8 @@ function install_dependencies_with_yum() {
     local common_dep="curl git gcc openresty-openssl111-devel unzip pcre pcre-devel openldap-devel"
     if [ "${1}" == "centos" ]; then
         # add APISIX source
-        sudo yum install -y https://repos.apiseven.com/packages/centos/apache-apisix-repo-1.0-1.noarch.rpm
+        local apisix_pkg=apache-apisix-repo-1.0-1.noarch
+        rpm -q --quiet ${apisix_pkg} || sudo yum install -y https://repos.apiseven.com/packages/centos/${apisix_pkg}.rpm
 
         # install apisix-base and some compilation tools
         # shellcheck disable=SC2086

